management/server/account_test.go

```diff
@@ -1943,72 +1943,104 @@ func TestAccount_AddJWTGroups(t *testing.T) {
 		Groups: map[string]*Group{
 			"group1": {ID: "group1", Name: "group1", Issued: GroupIssuedAPI, Peers: []string{}},
 		},
-		Settings: &Settings{JWTGroupsPropagationEnabled: true},
+		Settings: &Settings{GroupsPropagationEnabled: true},
 		Users: map[string]*User{
 			"user1": {Id: "user1"},
 			"user2": {Id: "user2"},
 		},
 	}
 
-	t.Run("don't add peers if no groups", func(t *testing.T) {
+	t.Run("api group already exists", func(t *testing.T) {
 		updated := account.AddJWTGroups("user1", []string{"group1"})
 		assert.False(t, updated, "account should not be updated")
-		assert.Empty(t, account.Groups["group1"].Peers, "group should not have any peers")
+		assert.Empty(t, account.Users["user1"].AutoGroups, "auto groups must be empty")
 	})
 
-	t.Run("add group without propagation", func(t *testing.T) {
-		account.Settings.JWTGroupsPropagationEnabled = false
+	t.Run("add jwt group", func(t *testing.T) {
 		updated := account.AddJWTGroups("user1", []string{"group1", "group2"})
 		assert.True(t, updated, "account should be updated")
 		assert.Len(t, account.Groups, 2, "new group should be added")
-		var group *Group
-		for _, g := range account.Groups {
-			if g.Name == "group2" {
-				group = g
-			}
-		}
-		assert.Empty(t, group.Peers, "group should not have any peers")
+		assert.Len(t, account.Users["user1"].AutoGroups, 1, "new group should be added")
+		assert.Contains(t, account.Groups, account.Users["user1"].AutoGroups[0], "groups must contain group2 from user groups")
 	})
 
-	t.Run("existed group update", func(t *testing.T) {
-		account.Settings.JWTGroupsPropagationEnabled = true
+	t.Run("existed group not update", func(t *testing.T) {
 		updated := account.AddJWTGroups("user1", []string{"group2"})
-		assert.True(t, updated, "account should be updated")
+		assert.False(t, updated, "account should not be updated")
 		assert.Len(t, account.Groups, 2, "groups count should not be changed")
```

Comment:
I don't know if that is good design making tests based on previous tests. I would prefer having each test independent of each other. If I change the order, or update the test t.Run("add jwt group") this would return completely wrong results.

Reply:
I didn't consider t.Run as separate tests, it just describes the steps which we do in the test. I could remove t.Run wrapper at all. Will it be more descriptive?

Reply:
Ah ok, no I thought those are different test cases. If we consider it one tests with different steps the it fine